src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -322,4 +322,69 @@ internal override object FromInternal(object obj)
             return obj;
         }
     }
+
+    /// <summary>
+    /// The data type for a SparseVectorUDT, used by SparseVector. 
+    /// </summary>
+    public class SparseVectorUDT : DataType
+    {
+        /// <summary>
+        /// Returns JSON object describing this type.
+        /// </summary>
+        internal override object JsonValue =>
+            new JObject(
+                new JProperty("type", "udt"),
+                new JProperty("class", "org.apache.spark.ml.linalg.VectorUDT"),
```

Comment:
Can you define this class under `Microsoft.Spark.ML.Linalg`?

Reply:
How is this type used? I don't see this being used in this PR?

Reply:
Will move it -or- it is used when calling CreateDataFrame to map the int,int,int[],double[] into a single column - a similar version appears in pyspark (not exactly the same) - if you think it is better for users to create it themselves I can take it out and document it instead?

Reply:
I think we can move it. Could you also add a test?